### DISCUSSION TITLE
How do we know OTPType before hand?

### DISCUSSION THREAD

**Discussion by pradeepb28** [supabase/supabase#18527] [Questions] [answered]
I am using Swift SDK, in that I am using SignInWithOTP using email to sign-up or sign-in. While this method works, when I am trying to verify it with OTP 6-digit code I should know OTPType if it is a signup user or already existing user. How can we know if it is a new user or existing user beforehand so that we can set appropriate OTPType in the method.

**Comment by GaryAustin1** [comment, marked as answer]
![image](https://github.com/supabase/supabase/assets/54564956/39e25957-6a61-4415-b00a-3625bef72da9)
https://github.com/supabase/gotrue/issues/905#issuecomment-1533128779

**Reply by landsman** [reply]
yesss, thank you! you saved me
